04-store-api/starter/controllers/products.js

```diff
@@ -0,0 +1,63 @@
+const Product = require('../models/product')
+
+
+const getAllProductsStatic = async (req,res) => {
+    const search = 'ab'
+    const products = await Product.find({price:{$gt: 30}}).sort('price').select('name price')
+    res.status(200).json({ products, ngHits: products.length })
+}
+const getAllProducts = async (req,res) => {
+    const { featured,company, name, sort, fields, numericFilters } = req.query
+    const queryObject = {}
+        if (featured){
+            queryObject.featured = featured === 'true' ? true : false
+        }
+        if (company){
+            queryObject.company = company 
+        }
+        if (name){
+            queryObject.name =  { $regex: name, $options: 'i' }
+        }
+        if (numericFilters){
+            const operatorMap = {
+                '>' : '$gt', 
+                '>=' : '$gte',
+                '=' :'$eq',
+                '<' : '$lt', 
+                '<=' : '$lte', 
+            }
+            const regEx = /\b(<|>|>=|=|<|<=)\b/g;
+            let filters = numericFilters.replace(regEx, (match)=> `-${operatorMap[match]}-`)
+            console.log(filters)
+        }
+
+    console.log(queryObject)
+    let result = Product.find(queryObject)
+    //sort
+    if(sort){
+        const sortList = sort.split(',').join(' ');
+        result = result.sort(sortList)
+    } 
+    else{
+        result.sort('createdAt')
+    }
+
+    if (fields){
+        const fieldsList = fields.split(',').join(' ');
+        result = result.select(fieldsList)
+    }
+    const page = Number(req.query.page) || 1
+    const limit = Number(req.query.limit) || 10
+    const skip = (page -1) * limit; 
```

Comment:
```suggestion
    const skip = (page - 1) * limit; 
```

nit: surround operators with whitespace